client/src/app/shared/hooks/useCreateEditModalState.ts

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { useContext, useState } from "react";
+
+export default function useCreateEditModalState<T>() {
+  const [modalState, setModalState] = React.useState<"create" | T | null>(null);
+  const isModalOpen = modalState !== null;
+  const resourceToUpdate = modalState !== "create" ? modalState : null;
+
+  const [resourceToDelete, setResourceToDelete] = React.useState<T | null>(
```

Comment:
The delete state doesn't belong here. Not part of CreateEdit.

Reply:
> Yes "instance" is user view where "tracker" is technical view. We have to maintain both views. The interface belongs more on the UI interface. The tracker is on the record side.

The only thing we need to maintain is the translation to user for form labels, descriptions, and column names. Makes more sense to use the same name for all of the UI components (forms/tables)  & underlying architecture (hooks, queries/ etc). Not sure what is so extreme about the change. Just helps when reasoning about the code.